3.0/m_samove.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2013, 2018-2019 Sadie Powell <sadie@witchery.services>
+ *   Copyright (C) 2013 Daniel Vassdal <shutter@canternet.org>
+ *   Copyright (C) 2012-2014, 2016 Attila Molnar <attilamolnar@hush.com>
+ *   Copyright (C) 2012, 2019 Robby <robby@chatbelgie.be>
+ *   Copyright (C) 2009 Daniel De Graaf <danieldg@inspircd.org>
+ *   Copyright (C) 2007, 2010 Craig Edwards <brain@inspircd.org>
+ *   Copyright (C) 2007 Robin Burchell <robin+git@viroteck.net>
+ *   Copyright (C) 2007 Dennis Friis <peavey@inspircd.org>
+ *   Copyright (C) 2006 jamie <jamie@e03df62e-2008-0410-955e-edbf42e46eb7>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/// $ModAuthor: Torben Hørup
+/// $ModAuthorMail: torben@t-hoerup.dk
+/// $ModDepends: core 3
+/// $ModDesc: Adds /SAMOVE command to move a user from one channel to another (basically combining SAPART+SAJOIN)
+
+#include "inspircd.h"
+
+/** Handle /SAMOVE
+ *
+ * Basically it's a SAPART + SAJOIN in 1 command
+ */
+class CommandSamove : public Command
+{
+ public:
+	CommandSamove(Module* Creator) : Command(Creator,"SAMOVE", 1)
+	{
+		allow_empty_last_param = false;
+		flags_needed = 'o';
+		syntax = "<nick> <fromchannel> <tochannel>";
+		TRANSLATE3(TR_NICK, TR_TEXT, TR_TEXT);		
+	}
+
+	CmdResult Handle(User* user, const Params& parameters) CXX11_OVERRIDE				
+	{
+		if (parameters.size() != 3) 
+		{
+			return CMD_FAILURE;			
+		}
+
+		const std::string& nickname = parameters[0];
+		const std::string& from_channel = parameters[1];
+		const std::string& to_channel = parameters[2];
+		std::string msg = std::string();
```

Comment:
Declaring msg here is unnecessary as you can use an empty string literal when calling PartUser.

Reply:
unfortunately Channel::PartUser() doesn't accept a const string reference ( i think thats a bug?)

Reply:
Ah, that is a bug yeah.

For now just move the declaration next to the function cal and remove the unnecessary value (i.e. `std::string msg;`).

Reply:
Thats also what i dit in my latest commit